providers/common/sql/src/airflow/providers/common/sql/operators/sql.py

```diff
@@ -1200,7 +1200,7 @@ def execute(self, context: Context):
             self.conn_id,
         )
         record = self.get_db_hook().get_first(self.sql, self.parameters)
-        if not record:
+        if record is None:
```

Comment:
Unrelated change? Best not to touch this file otherwise
```suggestion
        if not record:
```

Reply:
One of the tests that I modified failed because of handling a `None` record the same as a `False` record.
The relevant part is in `providers/common/sql/src/airflow/providers/common/sql/operators/sql.py`:

```python
        if record is None: # <- 1 
            raise AirflowException(
                "No rows returned from sql query. Operator expected True or False return value."
            )

        if isinstance(record, list):
            if isinstance(record[0], list):
                query_result = record[0][0]
            else:
                query_result = record[0]
        elif isinstance(record, tuple):
            query_result = record[0]
        else:
            query_result = record # <- 2 

        self.log.info("Query returns %s, type '%s'", query_result, type(query_result))

        follow_branch = None
        try:
            if isinstance(query_result, bool): # <- 3
                if query_result:
                    follow_branch = self.follow_task_ids_if_true
```